mail/dovecot/Portfile

```diff
@@ -1,110 +1,456 @@
+                    --disable-dependency-tracking
+}
+
+set perl5_major_version \
+                    5.28
+
+if {${name} eq ${subport}} {
+    depends_build-append \
+                    port:gettext \
+                    port:pkgconfig \
+                    port:wget
 
-checksums           sha1    457b16791a15e739c6bc8b02bc9c78f9ad794e39 \
-                    rmd160  59f619dba2aadef3cccd6ceb045a2e75a11700a2
+    depends_lib-append \
+                    port:bzip2 \
+                    port:libiconv \
+                    path:lib/libssl.dylib:openssl \
+                    port:xz \
+                    port:zlib
 
-depends_build       port:pkgconfig
-depends_lib         port:libiconv \
-                    port:zlib \
-                    path:lib/libssl.dylib:openssl
+    set default_internal_user   _dovecot
+    set default_login_user      _dovenull
+    add_users ${default_internal_user} group=${default_internal_user} realname=Dovecot
+    add_users ${default_login_user}    group=${default_login_user}    realname=Dovenull
 
-configure.args      --sysconfdir=${prefix}/etc/dovecot \
+    patch.pre_args  -p1
+    patchfiles      patch-doc-example-config-conf.d-10-master.conf.diff \
+                    patch-src-master-master-settings.c.diff
+
+    post-patch {
+        reinplace "s|@@default_internal_user@@|${default_internal_user}|g" \
+            ${worksrcpath}/doc/example-config/conf.d/10-master.conf
+        reinplace "s|@@default_login_user@@|${default_login_user}|g" \
+            ${worksrcpath}/doc/example-config/conf.d/10-master.conf
+    }
+
+    # Note: https://trac.macports.org/ticket/58544#comment:11 for this build logic
+    use_autoreconf  yes
+    autoreconf.cmd  ./autogen.sh
+    depends_build-append \
+                    port:autoconf \
+                    port:automake \
+                    port:libtool
+
+    pre-configure {
+        # the line `dummy < /dev/tty` breaks this; just copy the necessary file
+        # system -W ${worksrcpath} "gettextize -f"
+        xinstall -W ${worksrcpath} ${prefix}/share/gettext/config.rpath .
+    }
+
+    configure.args  --sysconfdir=${prefix}/etc \
                     --localstatedir=${prefix}/var \
-                    --with-ioloop=kqueue \
                     --with-ssl=openssl \
-                    --with-ssldir=${prefix}/etc/ssl \
-                    --enable-header-install
+                    --with-zlib \
+                    --with-bzlib \
+                    --with-ssldir=${prefix}/etc/openssl \
+                    --with-shared-libs \
+                    --with-pam
 
-configure.cppflags  -I${prefix}/include/openssl
+    configure.cppflags  -I${prefix}/include/openssl
+    configure.env-append \
+                    PANDOC=false
 
-# Create the dovecot user and group
-add_users dovecot group=dovecot realname=Dovecot
+    variant gssapi \
+        description "Enable GSSAPI authentication" {
+        configure.args-append       --with-gssapi=yes
+    }
```

Comment:
Since no dependencies are being added, why not always do this and just remove the variant?

Reply:
Sure.